applications/crossbar/src/modules/cb_configs.erl

```diff
@@ -15,198 +12,164 @@
-        {'error', _} ->
-            JObj = kz_doc:set_id(cb_context:req_data(Context), Id),
-            Context1 = cb_context:set_req_data(Context, JObj),
-            cb_context:validate_request_data(make_schema_name(Config), Context1)
+delete(Context, _ConfigId) ->
+    case strip_id(cb_context:doc(Context)) of
+        {[]} -> Context;
+        _ ->
+            flush(crossbar_doc:delete(Context, permanent))
     end.
 
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%% Load an instance from the database
-%% @end
-%%--------------------------------------------------------------------
--spec read(ne_binary(), cb_context:context()) -> cb_context:context().
-read(Config, Context) ->
-    Id = <<(?KZ_ACCOUNT_CONFIGS)/binary, Config/binary>>,
-    crossbar_doc:load(Id, Context, ?TYPE_CHECK_OPTION(<<"account_config">>)).
-
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%% Update an existing instance with the data provided, if it is
-%% valid
-%% @end
-%%--------------------------------------------------------------------
--spec update(ne_binary(), cb_context:context()) -> cb_context:context().
-update(Config, Context) ->
-    Id = <<(?KZ_ACCOUNT_CONFIGS)/binary, Config/binary>>,
-    validate_request_data(Id, Context).
-
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%% Update-merge partially an existing instance with the data provided, if it is
-%% valid
-%% @end
-%%--------------------------------------------------------------------
--spec validate_patch(ne_binary(), cb_context:context()) -> cb_context:context().
-validate_patch(Config, Context) ->
-    Id = <<(?KZ_ACCOUNT_CONFIGS)/binary, Config/binary>>,
-    crossbar_doc:patch_and_validate(Id, Context, fun validate_request_data/2).
-
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%% Validates existing instance
-%% @end
-%%--------------------------------------------------------------------
--spec validate_request_data(ne_binary(), cb_context:context()) -> cb_context:context().
-validate_request_data(<<"configs_", Config/binary>> = Id, Context) ->
-    OnSuccess = fun(C) -> crossbar_doc:load_merge(Id, C, ?TYPE_CHECK_OPTION(<<"account_config">>)) end,
-    cb_context:validate_request_data(make_schema_name(Config), Context, OnSuccess).
+-spec save_diff(cb_context:context(), path_token(), kz_json:object()) -> cb_context:context().
+save_diff(Context, ConfigId, Parent) ->
+    JObj = cb_context:doc(Context),
+    JObjDiff = kz_json:diff(JObj, Parent),
+    StoredDocument = kz_json:private_fields(kapps_account_config:get(cb_context:account_id(Context), ConfigId)),
+    Document = kz_json:merge_recursive(StoredDocument, JObjDiff),
+    flush(crossbar_doc:save(Context, Document, [])),
+    crossbar_doc:handle_datamgr_success(set_id(ConfigId, JObj), Context).
```

Comment:
> kapps_account_config would auto invalidate the cache when the doc is saved.

what do you mean by that? document is saved to db, kapps_account_config will try to get it from cache, and will get the previous version.